src/ORM/ResultSet.php

```diff
@@ -30,184 +28,8 @@
  * @template T of \Cake\Datasource\EntityInterface|array
  * @implements \Cake\Datasource\ResultSetInterface<T>
  */
-class ResultSet implements ResultSetInterface
+class ResultSet extends Collection implements ResultSetInterface
```

Comment:
I approve of extending Collection directly here. I'm interested in making Collection extensible and more customizable with collection iterators instead of CollectionInterface.

Not sure what ResultSetInterface means in the future, but I think this is ok for 5.1.

Reply:
> Not sure what ResultSetInterface means in the future, but I think this is ok for 5.1.

`ResultSetInterface` doesn't really hold much meaning. Maybe in the future we can just drop it and make `ORM\Query\SelectQuery:all()`'s return type as `Collection`/`CollectionInterface`.